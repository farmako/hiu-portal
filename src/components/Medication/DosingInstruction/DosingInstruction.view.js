import React from 'react';
import * as PropTypes from 'prop-types';
import DosageTiming from '../DosageTiming';
import DosingInstructionForPatient from '../DosingInstructionForPatient';
import DosingInstructionAsNeeded from '../DosingInstructionAsNeeded';

const DosingInstruction = (props) => {
  const { dosage } = props;
  return dosage ? (
    <div>
      <span>{dosage.text}</span>
      <br />
      <DosageTiming dosage={dosage} />
      <ul className="instruction-list-item">
        <DosingInstructionForPatient dosage={dosage} />
        <DosingInstructionAsNeeded dosage={dosage} />
      </ul>
    </div>
  ) : null;
};

DosingInstruction.propTypes = {
  dosage: PropTypes.shape({ text: PropTypes.string }),
};

DosingInstruction.defaultProps = {
  dosage: null,
};


export default DosingInstruction;
