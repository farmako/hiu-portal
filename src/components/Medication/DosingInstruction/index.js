import DosingInstruction from './DosingInstruction.view';

export default DosingInstruction;
