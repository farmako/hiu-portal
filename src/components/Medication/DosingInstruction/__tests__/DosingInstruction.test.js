import React from 'react';
import { shallow } from 'enzyme';
import DosingInstruction from '../DosingInstruction.view';

describe('DosingInstruction', () => {
  const mockData = {
    sequence: 1,
    text: '22/01/2020 to 07/02/2020',
    timing: {
      event: ['22/1/2019'],
    },
  };
  const wrapper = shallow(
    <DosingInstruction dosage={mockData} />,
  );

  it('should render properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should not render anything if data is undefined', () => {
    wrapper.setProps({ dosage: undefined });
    expect(wrapper.find('div').exists()).toEqual(false);
  });
});
