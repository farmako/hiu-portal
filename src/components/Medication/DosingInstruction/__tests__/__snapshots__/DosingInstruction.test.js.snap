// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DosingInstruction should render properly 1`] = `
<div>
  <span>
    22/01/2020 to 07/02/2020
  </span>
  <br />
  <DosageTiming
    dosage={
      Object {
        "sequence": 1,
        "text": "22/01/2020 to 07/02/2020",
        "timing": Object {
          "event": Array [
            "22/1/2019",
          ],
        },
      }
    }
  />
  <ul
    className="instruction-list-item"
  >
    <DosingInstructionForPatient
      dosage={
        Object {
          "sequence": 1,
          "text": "22/01/2020 to 07/02/2020",
          "timing": Object {
            "event": Array [
              "22/1/2019",
            ],
          },
        }
      }
    />
    <DosingInstructionAsNeeded
      dosage={
        Object {
          "sequence": 1,
          "text": "22/01/2020 to 07/02/2020",
          "timing": Object {
            "event": Array [
              "22/1/2019",
            ],
          },
        }
      }
    />
  </ul>
</div>
`;
